packages/r/intermarch3/goo-cli/internal/gnokey/executor.go

```diff
@@ -0,0 +1,279 @@
+package gnokey
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"goo-cli/internal/config"
+	"goo-cli/internal/utils"
+)
+
+// TxExecutor handles execution of gnokey transactions
+type TxExecutor struct {
+	KeyName   string
+	RealmPath string
+	ChainID   string
+	Remote    string
+	GasFee    string
+	GasWanted int64
+	Verbose   bool
+}
+
+// NewExecutor creates a new TxExecutor from config
+func NewExecutor(cfg *config.Config, verbose bool) *TxExecutor {
+	return &TxExecutor{
+		KeyName:   cfg.KeyName,
+		RealmPath: cfg.RealmPath,
+		ChainID:   cfg.ChainID,
+		Remote:    cfg.Remote,
+		GasFee:    cfg.GasFee,
+		GasWanted: cfg.GasWanted,
+		Verbose:   verbose,
+	}
+}
+
+// CallFunction executes a function call (transaction)
+func (e *TxExecutor) CallFunction(funcName string, args []string, sendCoins string) error {
+	// Build command arguments
+	cmdArgs := []string{
+		"maketx", "call",
+		"--pkgpath", e.RealmPath,
+		"--func", funcName,
+		"--gas-fee", e.GasFee,
+		"--gas-wanted", fmt.Sprintf("%d", e.GasWanted),
+		"--broadcast",
+		"--chainid", e.ChainID,
+		"--remote", e.Remote,
+	}
+
+	// Add function arguments
+	for _, arg := range args {
+		cmdArgs = append(cmdArgs, "--args", arg)
+	}
+
+	// Add coins if specified
+	if sendCoins != "" {
+		cmdArgs = append(cmdArgs, "--send", sendCoins)
+	}
+
+	// Add key name
+	cmdArgs = append(cmdArgs, e.KeyName)
+
+	// Always print the command for transactions (user needs to see what they're signing)
+	fmt.Println("Executing:")
+	printCommand("gnokey", cmdArgs)
+	fmt.Println()
+
+	// Execute the command with inherited stdin for interactive password input
```

Comment:
For the user, `goo` is an unknown program. It may be a security risk to expect the user to type their password into an unknown program. The idea is that `gnokey` is a known program which had a security review (keeps the password safe). It's ugly, but it may be necessary to copy the gnokey command to the clipboard and expect the user to paste it into a separate terminal (and type the password into `gnokey`). Need more comments from others.